datascience/tables.py

```diff
@@ -1202,23 +1246,21 @@ def sample(self, k=None, with_replacement=True, weights=None):
         """Returns a new table where k rows are randomly sampled from the
         original table.
 
-        Kwargs:
-            k (int or None): If None (default), all the rows in the table are
-                sampled. If an integer, k rows from the original table are
-                sampled.
+        Args:
+            ``k`` -- specifies the number of rows (``int``)  to be sampled from
+                self. Default is k is equal to number of rows in self.
 
-            with_replacement (bool): If True (default), samples the rows with
-                replacement. If False, samples the rows without replacement.
+            ``with_replacement`` -- (``boolean``), if true samples ``k`` rows
+                with replacement from self, else samples ``k`` rows without
+                replacement.
 
-            weights (list/array or None): If None (default), samples the rows
-                using a uniform random distribution. If a list/array is passed
-                in, it must be the same length as the number of rows in the
-                table and the values must sum to 1. The rows will then be
-                sampled according the the probability distribution in
-                ``weights``.
+            ``weights``: Array specifying valid probability distribution.
+                Rows in self are sampled according the the
+                probability distribution given by ``weights``. Default is
+                uniform distribution on [1, ... , n], n = number of rows.
```

Comment:
I actually think the previous description is clearer, since it specifies the format of `weights`. If you feel like yours is better, care to explain what you didn't like about the previous description?